File: 4.Programming course/C language/pwm.c
#include<stdio.h>
#include<wiringPi.h>
#include<softPwm.h>
#define PWM_Pin 1

int main()
{
	printf("This is an experiment of output PWM\n");
	wiringPiSetup();
	softPwmCreate(PWM_Pin,0,100);
	softPwmWrite(PWM_Pin,50);
	while(1)
	{

	}
	return 0;
}

File: 4.Programming course/Python language/pwm.py
import wiringpi

OUTPUT = 1
PIN_TO_PWM = 1

wiringpi.wiringPiSetup()
wiringpi.pinMode(PIN_TO_PWM,OUTPUT)
wiringpi.softPwmCreate(PIN_TO_PWM,0,100
wiringpi.softPwmWrite(PIN_TO_PWM,50)